common/network-common/src/test/java/org/apache/spark/network/server/OneForOneStreamManagerSuite.java

```diff
@@ -30,22 +31,47 @@
 public class OneForOneStreamManagerSuite {
 
   @Test
-  public void managedBuffersAreFeedWhenConnectionIsClosed() throws Exception {
+  public void testMissingChunk() {
     OneForOneStreamManager manager = new OneForOneStreamManager();
     List<ManagedBuffer> buffers = new ArrayList<>();
     TestManagedBuffer buffer1 = Mockito.spy(new TestManagedBuffer(10));
     TestManagedBuffer buffer2 = Mockito.spy(new TestManagedBuffer(20));
+    TestManagedBuffer buffer3 = Mockito.spy(new TestManagedBuffer(20));
+
     buffers.add(buffer1);
+    buffers.add(null);
     buffers.add(buffer2);
+    buffers.add(null);
+    buffers.add(buffer3);
 
     Channel dummyChannel = Mockito.mock(Channel.class, Mockito.RETURNS_SMART_NULLS);
-    manager.registerStream("appId", buffers.iterator(), dummyChannel);
-    assert manager.numStreamStates() == 1;
+    long streamId = manager.registerStream("appId", buffers.iterator(), dummyChannel);
+    Assert.assertEquals(1, manager.numStreamStates());
+    Assert.assertNotNull(manager.getChunk(streamId, 0));
+    Assert.assertNull(manager.getChunk(streamId, 1));
+    Assert.assertNotNull(manager.getChunk(streamId, 2));
+    manager.connectionTerminated(dummyChannel);
+    Mockito.verify(buffer1, Mockito.never()).release();
+    Mockito.verify(buffer2, Mockito.never()).release();
```

Comment:
why don't these get released?

Reply:
I have added a comment and released them manually at the `tearDown()`:
https://github.com/apache/spark/blob/737b14a63a9782af77ca6a273f9c55750aff11a3/common/network-common/src/test/java/org/apache/spark/network/server/OneForOneStreamManagerSuite.java#L73-L77